openedx/core/djangoapps/persistenttask/task.py

```diff
@@ -0,0 +1,63 @@
+"""
+Celery utility code for persistent tasks.
+"""
+
+from datetime import datetime
+import json
+
+from celery import Task
+import pytz
+import six
+
+from .models import FailedTask
+
+
+def serialize_args(args):
+    u"""
+    Combine arglist into a json string.
+    """
+    if not isinstance(args, (tuple, list)):
+        raise TypeError(u'Args must be a tuple or list')
+    return json.dumps(args)
+
+
+def serialize_kwargs(kwargs):
+    u"""
+    Combine kwarg dict into a json string
+    """
+    if not isinstance(kwargs, dict):
+        raise TypeError(u'Kwargs must be a dict')
+    return json.dumps(kwargs)
+
+
+# pylint: disable=abstract-method
+class PersistentTask(Task):
```

Comment:
I think I agree that PersistentTask doesn't explain the class's function well enough, but I'm not sure that PersistentFailedTask clears up the problem.  

At the points of use, it would look like the tasks are designed to fail:

```python
@task(base=PersistentFailedTask)
def my_task():
    pass
```

or 

```python
class MyTask(PersistentFailedTask):
    def run(self):
        pass
```

Why do we want FailedTasks?

I considered `RobustTask` as well, but I think robustness is not a specific enough quality.  `RetryableTask` is problematic, because we're specifically taking a different approach than celery's built-in retry mechanism.  `ReapplicableTask` is another option, that I kind of like.